lib/engine/game/g_1849/step/buy_e_token.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1849
+      module Step
+        class BuyEToken < Engine::Step::Base
```

Comment:
I expected this to be part of buy_train, not a separate step. Why is a separate step needed?